### ISSUE TITLE
bug(local-notifications) : No valid id supplied (TimedNotificationPublisher.java), notification is null

### ISSUE THREAD

**Issue by Houdhey** [ionic-team/capacitor-plugins#765]
### Plugin(s)
local-notifications

### Capacitor Version
3.3.1

### Platform(s)
Android

### Current Behavior
Trigger a notification in Firebase Console

### Expected Behavior
Do not crash the app

### Other Technical Details

**CLASS : TimedNotificationPublisher.java**
```  /**
   * Restore and present notification
   */
  public void onReceive(Context context, Intent intent) {
    NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
    Notification notification = intent.getParcelableExtra(NOTIFICATION_KEY);
    int id = intent.getIntExtra(LocalNotificationManager.NOTIFICATION_INTENT_KEY, Integer.MIN_VALUE);
    if (id == Integer.MIN_VALUE) {
      Logger.error(Logger.tags("LN"), "No valid id supplied", null);
    }
    notificationManager.notify(id, notification);
    rescheduleNotificationIfNeeded(context, intent, id);
  }
```

Here is my code sample : 
Push notifications don't display in foreground for Android. So I had to use Local Notifications inside the Push Notifications listener, to trigger a Local Notification and force the notification to be displayed in foreground Android.

Here is the code example : 
```
   // Listener for foreground app only
    PushNotifications.addListener('pushNotificationReceived', (push) => {

      // Foreground notification android using Local Notification
      if (this.platform.is('android')) {
        LocalNotifications.schedule({
          notifications: [
            {
              title: push.title,
              body: push.body,
              id: new Date().getUTCMilliseconds(),
              schedule: {
                at: new Date(Date.now() + 2000),
              },
              extra: push.data,
              channelId: 'pop-notifications',
              smallIcon: 'notification_icon',
              iconColor: '#ff6200',
            },
          ],
        }).then((result) => {
          
        });
```
When I log the notification, it's null because my id is not valid.
I don't get it.

**Comment by jcesarmobile** [comment]
Your issue is confusing, it's about local notifications but you mention firebase console, which is related to push notifications.
Can you provide a sample app and steps to reproduce?

**Comment by Houdhey** [comment]
Sorry I should explain the context first.
Push notifications don't display in foreground for Android. So I had to use Local Notifications inside the Push Notifications listener, to trigger a Local Notification and force the notification to be displayed in foreground Android.

Here is the code example : 
```
   // Listener for foreground app only
    PushNotifications.addListener('pushNotificationReceived', (push) => {

      // Foreground notification android using Local Notification
      if (this.platform.is('android')) {
        LocalNotifications.schedule({
          notifications: [
            {
              title: push.title,
              body: push.body,
              id: new Date().getUTCMilliseconds(),
              schedule: {
                at: new Date(Date.now() + 2000),
              },
              extra: push.data,
              channelId: 'pop-notifications',
              smallIcon: 'notification_icon',
              iconColor: '#ff6200',
            },
          ],
        }).then((result) => {
          
        });
```

Or if you can tell me that it's actually possible to get foreground notifications in Android using Push Notifications, but I've already looked for this problem, and this code is the only solution I had  to force having foreground notifications for Android.

So that's why I'm using firebase console to trigger a Push Notification, detect it on my Push Notification listener, and trigger a Local Notification using its data

**Comment by jcesarmobile** [comment]
I think you can make push notifications to appear by setting an importance of 4 or greater, there should be an importance field in firebase console

**Comment by Houdhey** [comment]
> I think you can make push notifications to appear by setting an importance of 4 or greater, there should be an importance field in firebase console

I tried everything, but no solutions found. This problem has been already asked here : 
https://github.com/ionic-team/capacitor-plugins/issues/765#issuecomment-1016241250

I'm going to wait capacitor release for Android 12

**Comment by jcesarmobile** [comment]
Have you tried creating a new notification channel with importance 4 or greater and send the notification to that channel? That should work

The link you have provided just points to this issue, so not sure what issue you mean when you said it’s been asked already.

**Comment by Houdhey** [comment]
Sorry wrong link.  

https://github.com/ionic-team/capacitor/discussions/3286

https://stackoverflow.com/questions/70368390/need-help-handling-foreground-notifications-on-android-in-my-capacitor-app


I created my channel like that : 
```
 const notificationChannel: NotificationChannel = {
          id: 'pop-notifications', // id must match android/app/src/main/res/values/strings.xml's default_notification_channel_id
          name: 'Pop notifications',
          description: 'Pop notifications',
          importance: 5,
          visibility: 1,
        };

PushNotifications.createChannel(notificationChannel);
```

I receive the notification in my PushNotifications.addListener('pushNotificationReceived') : 
{"id":"0:1642670604584058%dcdda047dcdda047","data":{},"title":"aaaaa","body":"bbbbb"}.  (Firebase Console)

I receive it only on the listener, it doesn't display when App is in foreground Android.

That is why I had to use LocalNotifications to force scheduling it. But not working for Android 12 now.
By the way, I tried using Ionic Local Notifications plugin, and it worked, but I don't want to mix plugins, and have a great Capacitor preference.

But if you tell me it's possible to use only Push Notifications to handle foreground/background, any platform, I take it
